index.js

```diff
@@ -46,77 +47,69 @@ const SCHEMA_SMTP_CONFIG = Joi.object()
+            Array.isArray(buildData.settings.email)) {
+            buildData.settings.email = {
+                addresses: buildData.settings.email,
+                statuses: DEFAULT_STATUSES
+            };
+        }
 
-                if (typeof buildData.settings.email === 'string' ||
-                    Array.isArray(buildData.settings.email)) {
-                    buildData.settings.email = {
-                        addresses: buildData.settings.email,
-                        statuses: DEFAULT_STATUSES
-                    };
-                }
+        if (!buildData.settings.email.statuses.includes(buildData.status)) {
+            return;
+        }
 
-                if (!buildData.settings.email.statuses.includes(buildData.status)) {
-                    return resolve(null);
-                }
-
-                const subject = `${buildData.status} - Screwdriver ${buildData.pipelineName} ` +
-                    `${buildData.jobName} #${buildData.buildId}`;
-                const message = `Build status: ${buildData.status}` +
-                    `\nBuild link:${buildData.buildLink}`;
-                const html = tinytim.renderFile(path.resolve(__dirname, './template/email.html'), {
-                    buildStatus: buildData.status,
-                    buildLink: buildData.buildLink,
-                    buildId: buildData.buildId,
-                    statusColor: COLOR_MAP[buildData.status]
-                });
+        const subject = `${buildData.status} - Screwdriver ${buildData.pipelineName} ` +
+            `${buildData.jobName} #${buildData.buildId}`;
+        const message = `Build status: ${buildData.status}` +
+            `\nBuild link:${buildData.buildLink}`;
+        const html = tinytim.renderFile(path.resolve(__dirname, './template/email.html'), {
+            buildStatus: buildData.status,
+            buildLink: buildData.buildLink,
+            buildId: buildData.buildId,
+            statusColor: COLOR_MAP[buildData.status]
+        });
 
-                const mailOpts = {
-                    from: this.config.from,
-                    to: buildData.settings.email.addresses,
-                    subject,
-                    text: message,
-                    html
-                };
+        const mailOpts = {
+            from: this.config.from,
+            to: buildData.settings.email.addresses,
+            subject,
+            text: message,
+            html
+        };
 
-                const smtpConfig = {
-                    host: this.config.host,
-                    port: this.config.port
-                };
+        const smtpConfig = {
+            host: this.config.host,
+            port: this.config.port
+        };
 
-                return resolve(emailer(mailOpts, smtpConfig));
-            });
-        });
+        // eslint-disable-next-line no-new
+        new Promise(resolve => resolve(emailer(mailOpts, smtpConfig)));
```

Comment:
If you want to preserve the previous spec, you should return the Promise object.
I think there is no need to wrap them in Promise since the base class would never use the returned object.
https://github.com/screwdriver-cd/notifications-base/blob/master/index.js#L31